### ISSUE TITLE
Can't build github.com/docker/docker/pkg/system on freebsd

### ISSUE THREAD

**Issue by fsouza** [moby/moby#42849]
**Description**

This is an issue that was reported in fsouza/go-dockerclient#878: when trying to build go-dockerclient on freebsd, users see an error where `github.com/docker/docker/pkg/system` fails to build (it's a transient dependency of go-dockerclient).

Not sure if yall care about it or not, but I figured I could open an issue asking whether you'd take a PR before sending one or simply vendoring & fixing on my side.

Thank you very much!

**Steps to reproduce the issue:**

Clone go-dockerclient and try to build it with `GOOS=freebsd`:

```
% git clone --depth=1 https://github.com/fsouza/go-dockerclient.git
% cd go-dockerclient
% GOOS=freebsd go build 
```

**Describe the results you received:**

Compilation fails with the following error message:

```
# github.com/docker/docker/pkg/system
../../go/pkg/mod/github.com/docker/docker@v20.10.8+incompatible/pkg/system/mknod.go:12:19: cannot use dev (type int) as type uint64 in argument to unix.Mknod
```

**Describe the results you expected:**

Successful build.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
(paste your output here)
```

**Output of `docker info`:**

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by thaJeztah** [comment]
Thanks for reporting; we indeed don't test on freebsd (there were some efforts on a *bsd implementation in the past, but those never reached completion), so code to support it may have been removed over time. That said, this looks like a small change to make it works (if this is the only code that's causing an issue), so feel free to contribute a PR.

I don't think we'll consider such a patch important enough to backport but at least it would make it into the next release.

From the linked issue;

> Yeah I looked into it a bit, it looks like we get it via github.com/docker/docker/pkg/archive.

I also had a quick look; I know the `pkg/system` package's scope is a bit large, so if we'd be able to trim it down a bit, I'd be 👍 on doing so, but it looks like `pkg/archive` uses quite a few bits of it, so that didn't seem like a good solution for this case.

One possible alternative is to use the same approach as containerd for the mknod; it appears to be used by `pkg/archive` (` handleTarTypeBlockCharFifo()`)  _only_ (no other bits of the codebase are using it), so perhaps we could move the functionality there; this is the equivalent code in pkg/archive in containerd (through https://github.com/containerd/containerd/commit/b4592091881e91d9f1fc62564b7842c89044f1df); https://github.com/containerd/containerd/blob/6fa958853197af31ffd1225a6c300589e5068667/archive/tar_unix.go#L101-L113

I'm ok with that being done in a follow-up though, if you prefer a smaller change.
